### PR TITLE
add buildConfigs support

### PR DESCRIPTION
**Pull request by containscafeine**
- [ ] add docs
- [x] add tests
- [ ] add examples

### CODE REVIEW

**Comment by kadel** [comment]
fixes https://github.com/kedgeproject/kedge/issues/385, right?

File: pkg/spec/types.go [hunk 1]
```diff
@@ -169,6 +178,10 @@ type ControllerFields struct {
 	// ref: io.kedge.SecretSpec
 	// +optional
 	Secrets []SecretMod `json:"secrets,omitempty"`
+	// List of OpenShift BuildConfigs
+	// ref: io.kedge.BuildConfigSpec
+	// +optional
+	BuildConfigs []BuildConfigSpecMod
```
Comment (containscafeine) [inline pkg/spec/types.go]: add tag here

**Comment by containscafeine** [comment]
#445 should go in first

File: pkg/spec/types.go (hunk 1 above)
Comment (containscafeine) [reply pkg/spec/types.go]: done

[the remark this answers is not in the document]
Comment (containscafeine) [reply pkg/spec/types.go]: done

**Comment by containscafeine** [comment]
sigh, I have no idea why I deleted this

**Comment by containscafeine** [comment]
reopened in #466

**Pull request closed without merging**